src/Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs

```diff
@@ -174,7 +173,8 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
                         EndpointReference? httpEndPoint = null;
                         if (resource is IResourceWithEndpoints resourceWithEndpoints)
                         {
-                            httpEndPoint = resourceWithEndpoints.GetEndpoint("http");
+                            var endpointName = sidecarOptions?.AppProtocol ?? "http";
```

Comment:
This coupling between the dapr protocol and the endpoint name of the app is interesting. Maybe we should have 2 separate options and a default fallback where we assume the app protocol and endpoint name are the same.

Reply:
@davidfowl you mean something like that:

```csharp
builder.AddProject<Projects.DaprServiceA>("servicea", "https")
       .WithDaprSidecar(new DaprSidecarOptions{
              AppEndpoint = "MyEndpoint",
              AppProtocol = "HTTP"
       })
```

And there are the following cases:

- `AppEndpoint` and `AppProtocol ` is `null`
  - use `http` for both
- `AppEndpoint` is **not** `null` and `AppProtocol` is `null`
  - use as the `AppProtocol` the protocol that is defined by the `AppEndpoint` URL. 
  - if the extraction failes, fall back to `http`
- `AppEndpoint` is `null` and `AppProtocol` is **not** `null`
  - Get the app endpoint with the `AppProtocol` name and set it as `AppEndpoint`
  - if the extraction failes, fall back to `http` endpoint
    - if that fails exception??
- `AppEndpoint` is not `null` and `AppProtocol` is not `null`
  - use both as they are defined

Reply:
resolved